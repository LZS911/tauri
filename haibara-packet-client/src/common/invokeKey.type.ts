export type invokeType = 'mini_grep';

interface InvokeArgs {
  [key: string]: unknown;
}
