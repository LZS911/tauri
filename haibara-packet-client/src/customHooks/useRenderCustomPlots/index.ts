import { Options } from '@ant-design/plots';
import useLanguage from '../useLanguage';
import useTheme, { getCurrentColorSchemeStrings } from '../useTheme';

const useRenderCustomPlots: () => Pick<
  Options,
  'color' | 'locale' | 'theme'
> & { className: string } = () => {
  const { isDark, currentColorScheme, currentThemeMode } = useTheme();
  const { currentLanguage } = useLanguage();
  const [color] = getCurrentColorSchemeStrings(isDark, currentColorScheme);
  const className =
    'rounded-md border border-stone-200 p-5 dark:border-stone-700 dark:bg-darkMode bg-white';

  return {
    color,
    className,
    locale: currentLanguage,
    theme: currentThemeMode,
  };
};

export default useRenderCustomPlots;
