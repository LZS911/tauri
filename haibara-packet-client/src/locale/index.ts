import zhCN from './zh-CN';
import enUS from './en-US';
import * as i18n from 'i18next';
import { initReactI18next } from 'react-i18next';
import LocalStorageWrapper from '../utils/LocalStorageWrapper';
import { Dictionary } from '../typing/common.type';
import CONSTANT from '../common/constant';

enum SupportLanguage {
  zhCN = 'zh',
  enUS = 'en',
}

i18n.use(initReactI18next).init({
  resources: {
    [SupportLanguage.zhCN]: zhCN,
    [SupportLanguage.enUS]: enUS,
  },
  lng: LocalStorageWrapper.getOrDefault(
    CONSTANT.LANGUAGE_STORAGE_KEY,
    SupportLanguage.zhCN
  ),
  fallbackLng: SupportLanguage.zhCN,
  interpolation: {
    escapeValue: false,
  },
});

const translation = (key: string, dic?: Dictionary) => {
  return i18n.t(key, dic);
};

export { SupportLanguage, translation };
export default i18n;
