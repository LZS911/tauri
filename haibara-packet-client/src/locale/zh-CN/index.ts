import grep from './grep';
import anomaly from './anomaly';
import dashboard from './dashboard';
import unit from './unit';
import backend from './backend';
import layout from './layout';
import common from './common';
import router from './router';
import login from './login';
export default {
  translation: {
    login: login,
    router: router,
    common: common,
    layout: layout,
    backend: backend,
    unit: unit,
    dashboard: dashboard,
    anomaly: anomaly,
    grep: grep,
  },
};
