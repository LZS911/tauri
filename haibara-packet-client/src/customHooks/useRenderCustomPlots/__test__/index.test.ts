import { renderHook } from '@testing-library/react';
import { useDispatch, useSelector } from 'react-redux';
import { ColorSchemeEnum, ThemeModeEnum } from '../../../common/enum';
import { SupportLanguage } from '../../../locale';
import useRenderCustomPlots from '../index';

jest.mock('react-redux', () => {
  return {
    ...jest.requireActual('react-redux'),
    useSelector: jest.fn(),
    useDispatch: jest.fn(),
  };
});

describe('test useRenderCustomPlots', () => {
  const useSelectorMock: jest.Mock = useSelector as jest.Mock;
  const useDispatchMock: jest.Mock = useDispatch as jest.Mock;

  beforeEach(() => {
    useDispatchMock.mockImplementation(() => {});
    useSelectorMock.mockImplementation((selector) => {
      return selector({
        userConfig: {
          lang: SupportLanguage.enUS,
          currentThemeMode: ThemeModeEnum.Light,
          currentColorScheme: ColorSchemeEnum.Blue,
        },
      });
    });
  });
  test('should match snapshot', () => {
    const { result } = renderHook(() => useRenderCustomPlots());

    expect(result.current).toMatchSnapshot();
  });
});
