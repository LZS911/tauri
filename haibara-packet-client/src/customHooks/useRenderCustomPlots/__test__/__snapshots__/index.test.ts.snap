// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test useRenderCustomPlots should match snapshot 1`] = `
Object {
  "className": "rounded-md border border-stone-200 p-5 dark:border-stone-700 dark:bg-darkMode bg-white",
  "color": "#1890ff",
  "locale": "en-US",
  "theme": "light",
}
`;
